### ISSUE TITLE
Bug: Radio Button's not Binding to Nested Properties

### ISSUE THREAD

**Issue by jacurtis** [rafaelpimpa/buefy#366]
### Overview of the problem

**Buefy** version: 0.5.4
**Vuejs** version: 2.4.4
**OS/Browser**: Mac OS Sierra/ Chrome

### Description

I noticed a bug when using the radio buttons as demonstrated here: https://buefy.github.io/#/documentation/radio (the physical buttons as opposed to the bullets)

In my testing I was not able to get them to work properly while using `v-bind` to bind them to a nested property such as `foo.bar`. If i pulled the property up to the top level such as `fooBar` then it would bind correctly, but moving it into a nested property caused issues.

Specifically, the radios seem to be able to set the nested property, but the other buttons would not deselect as the property changed to another radio in the group.

They were all bound to the same `v-model` property, i went as far as to copy the code on the documentation page exactly, and just moved the bound property from the base layer to a nested layer. It would work and not work as it moved in and out of being nested in another property.

### Steps to reproduce

1. Copy code for the radio buttons on this page: https://buefy.github.io/#/documentation/radio
1. Move the bound property from `radioButton` (such as the example given in the documentation) to `radio.button`
1. Refresh the page for the code to take effect
1. If you set a default value for the nested property you will notice that the radio buttons do not acknowledge the value and no buttons will be selected upon a refresh
1. Click and change the selected radio
    1. Interestingly, the value for the nested `radio.button` will update if you inspect it with each click of a radio
    1. However, the buttons will never "deselect" and once a radio button has been selected once, it can not set the property again (assuming you try to switch away and back to that radio)
1. Now create a property at the base level `radioButton` and you will see that this works as intended.

**Just to recap.**  
Given the example below. The radios bind correctly to `radioButton` property (because it is top level) but do not bind correctly to `radio.button` because it is nested.

```js
var app = new Vue({
    el: '#app',
    data: {
        radioButton: 'foo',     // good to go!
        radio: {
            button: 'foo'       // no bueno :( 
        }
    }
});
```

### Expected behavior

Radio button will select and deselect as other buttons bound to the same property are pressed

### Actual behavior

Buttons can be selected but not de-selected 😢

**Comment by jtommy** [comment]
@jacurtis `bRadioButton` works like native radio button and when it's selected is not possible unselect it.  As workaround, you could attach '@click.native' and if the `v-model` data is the same then set it to default value or empty. For example: 
```
<b-radio-button @click.native="unselect" v-model="radioButton" ...
...
unselect(event) {
   if (this.radioButton === event.target.value) {
       this.radioButton = ''
   }
}
```

**Comment by rafaelpimpa** [comment]
Not sure if it's your issue but something about nested properties on ``v-model`` has been fixed on vue 2.5: https://github.com/vuejs/vue/issues/5932#issuecomment-334040085

**Comment by jacurtis** [comment]
You are right, this is fixed now 😄
